twisted/conch/ssh/transport.py

```diff
@@ -1230,6 +1236,73 @@ def ssh_KEXINIT(self, packet):
                 self.ignoreNextPacket = True # Guess was wrong
 
 
+    def _ssh_KEX_ECDH_INIT(self, packet):
+        """
+        Called from ssh_KEX_DH_GEX_REQUEST_OLD.
+ 
+        Payload::
+
+            string client Elliptic Curve Diffie-Hellman public key
+
+        First we load the host's public/private keys.
+        Then we generate the ECDH public/private keypair for the given curve.
+        With that we generate the shared secret key.
+        Then we compute the hash to sign and send back to the client
+        Along with the server's public key and the ECDH public key.
+
+        @type packet: L{bytes}
+        @param packet: The message data.
+
+        @return: None.
+        """
+        #Get the raw client public key.
+        pktPub, packet = getNS(packet)
+
+        #Get the host's public and private keys
+        pubHostKey = self.factory.publicKeys[self.keyAlg]
+        privHostKey = self.factory.privateKeys[self.keyAlg]
+
+        #Create the first part of the exchange hash
+        h = _kex.getHashProcessor(self.kexAlg)()
+        h.update(NS(self.otherVersionString))
+        h.update(NS(self.ourVersionString))
+        h.update(NS(self.otherKexInitPayload))
+        h.update(NS(self.ourKexInitPayload))
+        h.update(NS(pubHostKey.blob()))
+        h.update(NS(pktPub))
+
+        #Get the base curve info
+        try:
+            shortKex = re.search("(nist[kpbt]\d{3})$", self.kexAlg).group(1)
+        except:
```

Comment:
You should be expecting a specific type of exception from this regular expression, not a blanket `except`.  For example, let's say a future maintainer comes through and accidentally changes `self.kexAlg` to `self.kxeAlg`; now this would raise an `AttributeError`, which would tell us `UnsupportedAlgorithm`.  In general you should catch the most specific exception type possible :).